skrifa/src/color/mod.rs

```diff
@@ -449,6 +453,108 @@ impl<'a> ColorGlyphCollection<'a> {
     }
 }
 
+/// A single color palette.
+pub struct ColorPalette<'a> {
+    cpal: cpal::Cpal<'a>,
+    /// Preparsed subarray of color records for just this palette.
+    sub_array: &'a [cpal::ColorRecord],
+    /// This palette's index in the CPAL table.
+    index: u16,
+}
+
+impl<'a> ColorPalette<'a> {
+    /// Returns the color at this palette index.
+    pub fn get(&self, color_index: u16) -> Option<cpal::ColorRecord> {
+        self.sub_array.get(usize::from(color_index)).copied()
+    }
+
+    /// Returns this palette's type flags (currently, whether this palette is appropriate for use on
+    /// a light and/or dark background). This may not always be present.
+    pub fn palette_type(&self) -> Option<cpal::PaletteType> {
+        self.cpal
+            .palette_types_array()?
+            .ok()?
+            .get(usize::from(self.index))
+            .map(|p| p.get())
+    }
+
+    /// Returns this palette's label/name, if present.
+    pub fn label(&self) -> Option<NameId> {
+        self.cpal
+            .palette_labels_array()?
+            .ok()?
+            .get(usize::from(self.index))
+            .and_then(|p| {
+                let name_id = p.get();
+                Some(name_id).filter(|name_id| name_id.to_u16() != 0xFFFF)
+            })
+    }
+
+    /// Returns this palette's index in the CPAL table.
+    pub fn index(&self) -> u16 {
+        self.index
+    }
+}
+
+/// Collection of color palettes for color glyphs.
+pub struct ColorPaletteCollection<'a> {
```

Comment:
```suggestion
pub struct ColorPalettes<'a> {
```

I've avoided the `Collection` suffix in recent additions (`GlyphNames` and `BitmapStrikes`) so let's do the same here.